api_blogs/lib/api_blogs_web/controllers/post_controller.ex

```diff
@@ -11,15 +12,29 @@ defmodule ApiBlogsWeb.PostController do
     render(conn, "index.json", posts: posts)
   end
 
-  def create(conn, %{"post" => post_params}) do
-    with {:ok, %Post{} = post} <- Blog.create_post(post_params) do
+  def create(conn, %{"post" => %{"content" => content, "title" => title}}) do
```

Comment:
Esse campo `published` é utilizado para gravar a data de criação de uma postagem?
Se sim, ao adicionar `timestamps` como foi feito no `schema` já são criados dois campos que são gerenciados automaticamente pelo Ecto.

```elixir
 schema "posts" do
    field :content, :string
    field :published, :naive_datetime
    field :title, :string
    field :updated, :naive_datetime
    field :user_id, :id

    timestamps()
  end
```

esses dois campos são `created_at` e `updated_at`.  Ambos são preenchidos automaticamente ao usar as funções `create` e `update` do `repo`.

Então talvez não seja necessário construir essa estrutura `new_post` assim no controller. Uma alternativa é adicionar apenas o `user_id` que foi pego.

```suggestion
  def create(conn, %{"post" => post_attrs}) do
  {:ok, %{"sub" => id}} = UserController.extract_id(conn)
    new_post = Map.put(post_attrs, "user_id", id)

    with {:ok, %Post{} = post} <- Blog.create_post(new_post) do
      conn
      |> put_status(:created)
      |> render("create.json", post: post)
    end
  end
```

Reply:
Sim, os campos `published` e `updated` registram os momentos em que o `post` foi respectivamente criado e atualizado pela última vez. Realmente, do jeito que eu tinha feito fica uma informação redundante. Se os campos criados com `timestamps()` são gerenciados automaticamente pelo `Ecto`, melhor ainda! Obrigada pela sugestão.